### PR TITLE
Fix `visitWhenLiteral`

### PR DESCRIPTION
**Pull request by eklimo**
`visitWhenLiteral` does not translate integer & string literals correctly.

#### Example input
```apex
switch on (x) {
  when 'a' {}
  when 1, - 1 {}
}
```

#### Current AST output
```jsonc
{
  "@type": "SwitchStatement",
  "condition": { /* ... */ },
  "whenClauses": [
    {
      "@type": "WhenValue",
      "values": [
        {
          "@type": "StringVal",
          "value": "'a'" // single quotes included
        }
      ],
      "statement": { /* ... */ }
    },
    {
      "@type": "WhenValue",
      "values": [
        {
          "@type": "IntegerVal",
          "value": 1
        },
        {
          "@type": "IntegerVal",
          "value": 1 // not negative
        }
      ],
      "statement": { /* ... */ }
    }
  ]
}
```

#### Adjusted AST output
```jsonc
{
  "@type": "SwitchStatement",
  "condition": { /* ... */ },
  "whenClauses": [
    {
      "@type": "WhenValue",
      "values": [
        {
          "@type": "StringVal",
          "value": "a" // correct value
        }
      ],
      "statement": { /* ... */ }
    },
    {
      "@type": "WhenValue",
      "values": [
        {
          "@type": "IntegerVal",
          "value": 1
        },
        {
          "@type": "UnaryExpression", // negative number represented by NEGATION operator to match normal expression structure
          "value": {
            "@type": "IntegerVal",
            "value": 1
          },
          "op": "NEGATION"
        }
      ],
      "statement": { /* ... */ }
    }
  ]
}
```

### CODE REVIEW

File: src/main/java/com/google/summit/translation/Translate.kt
Comment (aaronhurst-google) [inline src/main/java/com/google/summit/translation/Translate.kt:885]: Thank for catching this.

File: src/main/java/com/google/summit/translation/Translate.kt
Comment (aaronhurst-google) [inline src/main/java/com/google/summit/translation/Translate.kt:881]: You rightly observe that the parser separates the negative literals into 2 tokens and that whitespace is tolerated between them. I can't say much about the tradeoffs that led to this solution rather than lexing the negative number as a single token.

But on the AST side, I think this conflicts with the language intention: `when` values cannot be arbitrary expressions, and literals and enums and types are the only things allowed (or so the language reference claims).

https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_switch.htm

I guess I'm curious about whether Jorje creates a unary operator or a negative literal... but only a little.  :-) I'd still assert that a negative literal value is preferable. (If nothing else, it simplifies the patterns that code analysis needs to expect and handle.)

**Review by aaronhurst-google — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/summit/translation/Translate.kt
Comment (eklimo) [reply src/main/java/com/google/summit/translation/Translate.kt:881]: Yes, Jorje does interpret a negative integer in this context as having a negation operator, which is more consistent with how normal literal expressions are parsed. However, I do see your point that there is a more restricted set of valid values here; in that case, this difference would have to be handled in the PMD tree builder rather than directly in Summit AST. (It does bother me a little that negative literals will have two different representations in the AST format, though.) Should we go with this instead?

By the way, even without the `UnaryExpression` change, there was a bug here that mishandled the negative sign and still will need to be fixed.

**Review by zorzella — APPROVED** [review]
I defer to Aaron on the decision about the negative literal.

File: src/main/java/com/google/summit/translation/Translate.kt
Comment (aaronhurst-google) [reply src/main/java/com/google/summit/translation/Translate.kt:881]: You're right that the negative sign is not typically part of the literal, as least as far as the parser is concerned. I looked at a few languages that do have formal specifications (e.g. https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.10.1) and this is the case.

As far as the Apex language guide is concerned, probably their language around `when` values and literals is just a bit sloppy. I'll relax my reading of this as a precise guide about the language.

Back to the question at hand. Because the AST is an abstracted version of the parse tree, it need not be bound by the parser or its specification (even if it did exist). We have an open choice between a representation that is simpler or one that is closer to the syntactic structure.

If this were a compiler, we'd definitely want to rewrite `UnaryNegation(PositiveLiteral)` as `NegativeLiteral` during optimization. In static analysis, some checks do care about certain aspects of the syntax. Probably not this aspect; and anyway there isn't any information loss here.

I think that about sums it up. I'll leave the representational choice in your hands.

**Review by aaronhurst-google — APPROVED** [review]
LGTM pending decision on negative literals.

**Pull request merged**
